### DISCUSSION TITLE
Best way to clear all images on a listing

### DISCUSSION THREAD

**Discussion by simonj** [etsy/open-api#1186] [Q&A]
Hey guys,

What is the best way to clear all images on a listing?

Let's say I have a GUI like this where I upload 5 new images and remove the old ones

<img width="1292" alt="Skærmbillede 2024-01-04 kl  22 21 59" src="https://github.com/etsy/open-api/assets/183790/334b5d92-92b4-4719-b3c4-a588ae216809">

There is a rule that a listing needs to have at least 1 image. So I can't make an API call to get all the images and then for each them to delete them one by one.

I could start by adding 1 image and then in the same function that adds new images I delete all the images but then again something could happen that I would delete the first one that was just created, and what if the listing has all 10 images, then I can't start by adding 1 image because then there is too many.

It's so weird that we don't have a "clear image" API call or something better.

How do you guys handle that?

**Comment by showtroylove** [comment]
Those slot 1 - 10 are like array cells or a list.  You decide which slot to place an image being uploaded by using the Rank property. If the slot is filled you effectively are replacing that image, if empty adding.  

**Be sure to mark this comment as an answer if it helps you resolve your question.**
